modules/shared/facts.d/win_env_fact.ps1

```diff
@@ -0,0 +1,27 @@
+# Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Creates useful facts based off of system environment variables
+
+$systemdrive     = $env:systemdrive
+$system32        = "$env:systemdrive\\windows\\system32"
+$programdata     = $env:programdata.replace("\","\\")
+$programfiles    = $env:ProgramW6432.replace("\","\\")
+$programfilesx86 = "$systemdrive\\Program Files (x86)"
+
+# Environment variables
+write-host "systemdrive=$env:systemdrive"
```

Comment:
When writing custom facts, I would like the name to indicate so and also indicate if it is os specific.  For example, 'systemdrive' should be 'custom_win_systemdrive'

Reply:
Facts and applicable hiera values are now prefixed with custom_win.